### PR TITLE
Code refactoring tomas

### PR DESCRIPTION
**Pull request by tomaskrcka**
Changes:
- removed python script for generating credentials (now manual work)
- connecting to wifi and mqtt is state machine (one task)
- one new task for checking other tasks and watchdog triggering
- documentation
- publishing and subscribing is now compatible with firmware version 1.0.0
- logging
- removed submodules (be able to download project in zip)

### CODE REVIEW

**Comment by chainq** [comment]
I did a really quick review (no time for a detailed one due to BSH), to have an impression of the changes, looking good, 👍 from me.

File: src/tasks.c [hunk 1]
```diff
@@ -5,249 +5,206 @@
...
+            {
+                state = MQTT_CONN;
+            }
+            else
+            {
+                vTaskDelay(RECONNECT_PERIOD / portTICK_RATE_MS);
+            }
+            break;
+        case MQTT_CONN:
+            if (0 == MqttInit()) {
```
Comment (jamesmunns) [inline src/tasks.c]: Inconsistent brackets, please make consistent

File: src/tasks.c [hunk 2]
```diff
@@ -5,249 +5,206 @@
...
     }
+    return 0;
+}
+
+/*
+ * period - publish period in ms
+ */
+void Tasks_PublishPeriod(uint32_t period)
+{
+    if (period == 0)
```
Comment (jamesmunns) [inline src/tasks.c]: Consider always using `{}` for conditionals (minor style comment)

File: src/processcheck_task.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    int8_t nMaxState;
+    char name[configMAX_TASK_NAME_LEN];
+} proccheck_t;
+
+static proccheck_t abTaskFlags[PROCESSCHECK_MAXPROCS];
+static xTaskHandle mainTaskHandle;
+
+static void ProcessCheck_Task(void *context)
+{
+    uint8_t i = 0;
```
Comment (jamesmunns) [inline src/processcheck_task.c]: consider moving this to inside the for loop on line 34

File: src/processcheck_task.c [hunk 4]
```diff
@@ -0,0 +1,114 @@
+/*
+ * processcheck_task.c
```
Comment (jamesmunns) [inline src/processcheck_task.c]: It would be nice to have at least function level comments for this file.

File: src/tasks.c [hunk 5]
```diff
@@ -5,249 +5,206 @@
 #include "mqtt.h"
 #include "wifi.h"
 #include "controls.h"
+#include <FreeRTOS.h>
```
Comment (jamesmunns) [inline src/tasks.c]: Please add function level comments for this file (src/tasks.c)

File: documentation/xdk-onboarding.md [hunk 6]
```diff
@@ -141,6 +141,71 @@ JSON file | credentials.h
...
+
+**NOTE:** You must log in to the account where Vertex has been onboarded.
+All devices that will be associated with Vertex must be on the same account as
+Vertex itself.
+
+##### Parameters
+
+The following parameters are required in order to associate the device with Vertex:
+
+-  `Vertex Id`: UUID of your Vertex gateway. Once you're done with the section ["Configuration & Onboarding"](https://github.com/relayr/vertex/blob/master/docs/vertex_DSK_readme.md#configuration--onboarding), the gateway should show up as another device. This UUID simply corresponds to the "device ID" of the gateway, that can be displayed as described [here](http://docs.relayr.io/getting-started/devices-guide/#editing--deleting-a-device).
```
Comment (jamesmunns) [inline documentation/xdk-onboarding.md]: is `https://github.com/relayr/vertex` public?

File: documentation/xdk-onboarding.md [hunk 7]
```diff
@@ -141,6 +141,71 @@ JSON file | credentials.h
...
+-  `Authorization Token`: A token that authorizes you to make API calls to your user account and - devices that you manage. [Here is more info about it](http://docs.relayr.io/getting-started/account-creation/#user-id-and-authorization-token) and where to retrieve it.
+-  `Device Id`: UUID of the device that will be associated with Vertex. Provided by the relayr - platform [when adding a device](http://docs.relayr.io/getting-started/devices-guide/).
+-  `User`: MQTT user, in this case the same as the `Device Id`.
+-  `Password`: MQTT password. Provided by the relayr platform [when adding a device](http://docs.relayr.io/getting-started/devices-guide/).
+-  `MQTT Server Hostname`: The IP address of your Vertex gateway. This is not specifically required in this step, but it will be added into the code examples. For more information, please refer to the step #8 of the section ["Configuration & Onboarding"](https://github.com/relayr/vertex/blob/master/docs/vertex_DSK_readme.md#configuration--onboarding) as described on the Vertex documentation.
```
Comment (jamesmunns) [inline documentation/xdk-onboarding.md]: Same comment about vertex

**Pull request merged**